# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CompositorBridgeChild.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 gfx/layers/ipc/CompositorBridgeChild.cpp
--- a/gfx/layers/ipc/CompositorBridgeChild.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/gfx/layers/ipc/CompositorBridgeChild.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -13,6 +13,7 @@
 #include "base/task.h"           // for NewRunnableMethod, etc
 #include "mozilla/StaticPrefs_layers.h"
 #include "mozilla/dom/TabGroup.h"
+#include "mozilla/dom/WebGLChild.h"
 #include "mozilla/layers/CompositorManagerChild.h"
 #include "mozilla/layers/ImageBridgeChild.h"
 #include "mozilla/layers/APZChild.h"
@@ -1060,6 +1061,25 @@
   return true;
 }
 
+dom::PWebGLChild* CompositorBridgeChild::AllocPWebGLChild(
+    const WebGLVersion& aVersion, HostWebGLCommandSink&& aCommandSink,
+    HostWebGLErrorSource&& aErrorSource) {
+  MOZ_CRASH("Should create PWebGLChild outside of IPDL");
+  return nullptr;
+}
+
+dom::PWebGLChild* CompositorBridgeChild::AllocPWebGLChild(
+    const WebGLVersion& aVersion, const HostWebGLCommandSink& aCommandSink,
+    const HostWebGLErrorSource& aErrorSource) {
+  MOZ_CRASH("Should create PWebGLChild outside of IPDL");
+  return nullptr;
+}
+
+bool CompositorBridgeChild::DeallocPWebGLChild(PWebGLChild* aActor) {
+  delete static_cast<dom::WebGLChild*>(aActor);
+  return true;
+}
+
 void CompositorBridgeChild::WillEndTransaction() { ResetShmemCounter(); }
 
 PWebRenderBridgeChild* CompositorBridgeChild::AllocPWebRenderBridgeChild(